lib/models/Model.d.ts

```diff
@@ -234,7 +234,7 @@ export declare abstract class Model<T> extends Hooks {
    *
    * If you provide an `include` option, the number of matching associations will be counted instead.
    */
-  static count(options?: ICountOptions): Promise<number>;
+  static count<T extends Model<T>>(options?: ICountOptions<T>): Promise<number>;
```

Comment:
Can you add `this: (new () => T)` here as well? See `findAll` as an example: https://github.com/suttna/sequelize-typescript/blob/42240661d503d614781d1353f131a37db33300be/lib/models/Model.d.ts#L201